kolibri/core/analytics/pskolibri/common.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import
+from __future__ import division
+
+import functools
+import os
+import sys
+from collections import namedtuple
+
+PY3 = sys.version_info[0] == 3
+POSIX = os.name == "posix"
+WINDOWS = os.name == "nt"
+LINUX = sys.platform.startswith("linux")
+MACOS = sys.platform.startswith("darwin")
+
+if PY3:
+
+    def b(s):
+        return s.encode("latin-1")
+
+
+else:
+
+    def b(s):
+        return s
+
+
+ENCODING = sys.getfilesystemencoding()
+if not PY3:
+    ENCODING_ERRS = "replace"
+else:
+    try:
+        ENCODING_ERRS = sys.getfilesystemencodeerrors()  # py 3.6
+    except AttributeError:
+        ENCODING_ERRS = "surrogateescape" if POSIX else "replace"
+
+pcputimes = namedtuple('pcputimes', ['user', 'system', 'children_user', 'children_system'])
+
+
+class NoSuchProcess(Exception):
+    """Exception raised when a process with a certain PID doesn't
+    or no longer exists.
+    """
+
+    pass
+
+
+class AccessDenied(Exception):
+    """Exception raised when permission to perform an action is denied."""
+
+    pass
+
+
+def memoize(fun):
```

Comment:
This should exist elsewhere? Have you looked? Maybe in Django?

Reply:
Yes, I looked and didn't find it. 
This errors are for OS errors, I don't think Kolibri has faced them before.

Reply:
No, that's true -- now I looked, too.. was just pretty sure that this stuff used to be shipped. But Django also removed the bundled `lru_cache` not long ago.